### PR TITLE
Read requirement files recursively

### PR DESCRIPTION
**Pull request by Paebbels**
**Description**

This PR fixes issues in `setup.py` and it's the function to read dependencies from `requirements.txt`.

Since last PR for DOM, the `requirements.txt` has been split into multiple files, so users can choose what to install depending on the usecase. Common requirement files are included via `-r ...` option. The new read function can now handle these recursive includes. Same includes are also deduplicated.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -96,6 +99,8 @@ def get_version():
         namespace=gitHubNamespace, projectName=projectName
     )
 )
+requirements = list(dict.fromkeys(get_requirements(requirementsFile)))
```
Comment (Xiretza) [inline setup.py:102]: If the order of requirements is not significant (I don't think it is?), `list(set(get_requirements(requirementsFile)))` might be a bit clearer here.

File: setup.py [hunk 2]
```diff
@@ -63,8 +63,11 @@ def get_requirements(file: Path) -> List[str]:
     requirements = []
     with file.open("r") as fh:
         for line in fh.readlines():
-            if line.startswith("#"):
+            if line.startswith("#") or line.strip() == "":
```
Comment (umarcor) [inline setup.py:66]: It seems that we are doing the same strip operation on the same variable in three different places. What about `_line = line.strip()` right after L65, so we can then remove `strip()` from L66, L69 and L72?

File: setup.py [hunk 3]
```diff
@@ -63,8 +63,11 @@ def get_requirements(file: Path) -> List[str]:
...
                 continue
+            elif line.startswith("-r"):
+                filename = line[3:].strip()
```
Comment (umarcor) [inline setup.py:69]: If we move strip above, `line[3:]` can be written in L70 without variable `filename`.

File: setup.py (hunk 1 above)
Comment (Paebbels) [reply setup.py:102]: for new Python versions (so starting around 3.6), `list(set(...))` is not recommended anymore.

File: setup.py (hunk 2 above)
Comment (Paebbels) [reply setup.py:66]: No the strip operations are not identical.

Btw. don't name local variables `_...` (I saw this multiple times in your code.)
These are private members of classes.

File: setup.py (hunk 1 above)
Comment (Xiretza) [reply setup.py:102]: Interesting, do you have a source for that? I can't really imagine `dict.fromkeys` being more idiomatic than `set` even though there is no dict involved.

File: setup.py (hunk 3 above)
Comment (Paebbels) [reply setup.py:69]: There is no performance improvement by removing variables.

At first you remove documentation from code and you complicate reviews, as it's unclear what information is in the substring `[3:]`. At second you looking forward for a conflict with black, which will then cripple the code.

File: setup.py (hunk 1 above)
Comment (Paebbels) [reply setup.py]: Thanks @Xiretza. I did tests with `timeit.repeat(...)` myself and the articles about list deduplication are not correct.
A modified version has been force-pushed.

File: setup.py (hunk 3 above)
Comment (Paebbels) [reply setup.py:70]: @umarcor Oh there are 2 more strips ...

See update version.

**Comment by umarcor** [comment]
I rebased on top of master.

**Pull request merged**